metricbeat/mb/mb.go

```diff
@@ -94,6 +104,44 @@ func (m *BaseModule) UnpackConfig(to interface{}) error {
 	return m.rawConfig.Unpack(to)
 }
 
+// ReConfigure re-configures the module with the given raw configuration
+func (m *BaseModule) ReConfigure(config *common.Config, register *Register) error {
+	currName := m.name
+	currConfig := m.config
+	currRawConfig := m.rawConfig
+
+	var chkConfig struct {
+		Module string `config:"module"`
+	}
+	if err := config.Unpack(&chkConfig); err != nil {
+		return errors.Wrap(err, "error parsing new module configuration")
+	}
+
+	// Don't allow module name change
+	if chkConfig.Module != "" && chkConfig.Module != currName {
+		return fmt.Errorf("cannot change module name from %v to %v", currName, chkConfig.Module)
+	}
+
+	if err := config.SetString("module", -1, currName); err != nil {
+		return errors.Wrap(err, "unable to set existing module name in new configuration")
+	}
+
+	if err := config.Unpack(&m.config); err != nil {
```

Comment:
From this point on, we update `m`'s fields in this method. Note that this is not thread-safe. How critical is that? Should add a mutex to the `BaseModule` struct?

Reply:
Don't worry about this, I think there is no concurrency on modules instantiation.